Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraintPalmUp.cs

```
@@ -163,5 +203,209 @@ protected override bool IsValidController(IMixedRealityController controller)
+        /// </summary>
+        /// <param name="jointedHand"></param>
+        /// <param name="handPlane"></param>
+        /// <param name="activationPoint"></param>
+        /// <returns></returns>
+        private bool GenerateHandPlaneAndActivationPoint(IMixedRealityHand jointedHand, out Plane handPlane, out Vector3 activationPoint)
+        {
+            // Generate the hand plane that we're using to generate a distance value.
+            // This is done by using the index knuckle, pinky knuckle, and wrist
+            MixedRealityPose indexKnuckle;
+            MixedRealityPose pinkyKnuckle;
+            MixedRealityPose wrist;
+
+            if (jointedHand.TryGetJoint(TrackedHandJoint.IndexKnuckle, out indexKnuckle) &&
+                jointedHand.TryGetJoint(TrackedHandJoint.PinkyKnuckle, out pinkyKnuckle) &&
+                jointedHand.TryGetJoint(TrackedHandJoint.Wrist, out wrist))
+            {
+                handPlane = new Plane(indexKnuckle.Position, pinkyKnuckle.Position, wrist.Position);
+                activationPoint = handPlane.ClosestPointOnPlane(GenerateActivationPoint(jointedHand));
+                return true;
+            }
+            else // Otherwise, set the activation point and plane to default values and return false
+            {
+                handPlane = new Plane();
+                activationPoint = Vector3.zero;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// This function attempts to generate an activation point based on what the currently-selected safe zone is.
+        /// The activation point is a Vector3 that represents the "Center" point of the area that a user needs to gaze at to
+        /// activate the attached 
+        /// </summary>
+        /// <param name="jointedHand"></param>
+        /// <returns></returns>
+        private Vector3 GenerateActivationPoint(IMixedRealityHand jointedHand)
```

It's a little  surprising to me that GenerateHandPlaneAndActivationPoint returns true/false but this method doesn't. It seems like this method could fail the same way GenerateHandPlaneAndActivationPoint could, yet in this one the result is to set values to Vector3.zero instead of returning false in case of failure. Seems like it would be safer to use the pattern to do TryBlah and return false instead of failure. This can help debug tricky bugs in the future.

---

That's a fair assessment. I can change this.